vignettes/deepdive.Rmd

```diff
@@ -0,0 +1,389 @@
+
+```{r, eval = FALSE}
+adsl %>%
+  xportr_type(var_spec, "ADSL", "message") %>%
+  xportr_length(var_spec, "ADSL", "message") %>%
+  xportr_label(var_spec, "ADSL", "message") %>%
+  xportr_order(var_spec, "ADSL", "message") %>%
+  xportr_format(var_spec, "ADSL", "message") %>%
+  xportr_write("adsl.xpt", label = "Subject-Level Analysis Dataset")
+```
+
+To help reduce these repetitive calls, we have created the `xportr_metadata()` function. A user can just **set** the _metadata object_ and the Domain name in the first call and this will be passed onto the other functions.  Much cleaner!
+
+```{r, eval = FALSE}
+adsl %>%
+  xportr_metadata(var_spec, "ADSL") %>%
+  xportr_type() %>%
+  xportr_length() %>%
+  xportr_label() %>%
+  xportr_order() %>%
+  xportr_format() %>%
+  xportr_write("adsl.xpt", label = "Subject-Level Analysis Dataset")
+
+```
+
+
+## Warnings and Errors 
+
+For the next six sections, we are going to explore the Warnings and Errors messages generated by `{xportr}` functions. To better explore these, we will either manipulate the ADaM dataset or specification file to help showcase the ability of the `{xportr}` functions to detect issues.  
+
+**NOTE:** These datasets and specification are not available directly from the package. You can access them on our [repo](https://github.com/atorus-research/xportr) in the `example_data_specs` folder. This is to keep the package to a minimum size.  
+
+
+```{r}
+# options(xportr.variable_name = "variable", 
+#         xportr.label = "label",
+#         xportr.type_name = "type",
+#         xportr.format = "format",
+#         xportr.length = "length",
+#         xportr.order_name = "order")
+```
+
+### Setting up our metadata object
+
+First, lets read in the specification file and call it `var_spec`.  We will also do some slight manipulation to the columns names by doing all lower case and changing `Data Type` to `type. 
+
+```{r}
+var_spec <- readxl::read_xlsx(spec_loc, sheet = "Variables") %>% 
+  dplyr::rename(type = "Data Type") %>%
+  rlang::set_names(tolower)
+```
+
+```{r}
+columns2hide <- c("significant digits", "mandatory", "assigned value", "codelist", "common", "origin", "pages", "method", "predecessor", "role", "comment", "developer notes")
+
+datatable(
+  var_spec, rownames = FALSE,
+  extensions = 'Buttons', options = list(
+    dom = 'Bfrtip',
+    columnDefs = list(list(visible = FALSE, targets = columns2hide )))
+  )
+```
+
+### `xportr_type()`
+
+We are going to explore the type column in the metadata object. A submission to a Health Authority should only have character and numeric types in the data. In the  `ADSL` data we will have several columns that are in the Data type: `TRTSDT`, `TRTEDT`, `DISONSDT`, `VISIT1DT` and `RFENDT` and we will change one variable type to a factor variable. 
+
+```{r}
+adsl_loc <- here::here("example_data_specs", "adsl.xpt")
```

Comment:
I think you missed the `adsl.xpt` file when adding to git, probably due the entry on the `.gitignore` file

Reply:
whoops yeah that shouldn't be there!  it is actually from writing out the file